improver/regrid/grid.py

```diff
@@ -378,13 +378,8 @@ def create_regrid_cube(cube_array: ndarray, cube_in: Cube, cube_out: Cube) -> Cu
         Regridded result cube
     """
     # generate a cube based on new data and cube_in
-    cube_v = Cube(
-        cube_array,
-        standard_name=cube_in.standard_name,
-        var_name=cube_in.var_name,
-        units=cube_in.units,
-        attributes=cube_in.attributes,
-    )
+    cube_v = Cube(cube_array, units=cube_in.units, attributes=cube_in.attributes)
+    cube_v.rename(cube_in.var_name or cube_in.standard_name or cube_in.long_name)
```

Comment:
We've lost the explicit setting of the `var_name` with this change (though we seldom use it for cubes). An Iris rename command sets the var_name to `None` in every case. The order here also gives a preference to the var name over the standard or long names when setting the primary name, which seems a little odd.

```suggestion
    cube_v.rename(cube_in.standard_name or cube_in.long_name)
    cube_v.var_name = cube_in.var_name
```

Reply:
Thanks for the explanation. I wasn't totally clear on the issues regarding var_name and the .rename method. I've implemented your suggested change.